pkg/scheduling/volumeusage.go

```diff
@@ -126,22 +122,34 @@ func resolveDriver(ctx context.Context, kubeClient client.Client, volumeName str
 		if err != nil {
 			return "", err
 		}
-		if driverName != "" {
+		if driverName == "" {
 			return driverName, nil
 		}
 	}
-	if storageClassName != "" {
-		driverName, err := driverFromSC(ctx, kubeClient, storageClassName)
-		if err != nil {
-			return "", err
-		}
-		if driverName != "" {
-			return driverName, nil
+
+	// This can occur in two scenarios:
+	//  1. The storage class was explicitly set to "" to disable dynamic provisioning
+	//  2. The storage class was not set but the cluster doesn't have a default storage class
+	// In either of these cases, a PV must have been previously bound to the PVC and has since been removed. We can
+	// ignore this PVC while computing limits and continue.
+	if storageClassName == "" {
+		logging.FromContext(ctx).Errorf("failed to update volume limits for volume with unbound PVC, no storage class specified")
```

Comment:
```suggestion
		logging.FromContext(ctx).Errorf("failed updating volume limits for volume with unbound PVC, no storage class specified")
```

nit: style is to use the gerund